File: src/types/category.ts
export interface Category {
  id: string;
  name: string;
  path: string;
  createdAt: string;
  updatedAt: string;
  noteCount: number;
}

export interface CreateCategoryRequest {
  name: string;
}

export interface RenameCategoryRequest {
  id: string;
  newName: string;
}

export interface CategoryOrderConfig {
  categoryIds: string[];
}

File: src-tauri/src/models/category.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Category {
    pub id: String,
    pub name: String,
    pub path: String,
    pub created_at: String,
    pub updated_at: String,
    pub note_count: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct CreateCategoryRequest {
    pub name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RenameCategoryRequest {
    pub id: String,
    pub new_name: String,
}